ipam/allocator.go

```diff
@@ -416,32 +416,32 @@ func (a *Allocator) getPredefinedPool(as string, ipV6 bool) (*net.IPNet, error)
 }
 
 // RequestAddress returns an address from the specified pool ID
-func (a *Allocator) RequestAddress(poolID string, prefAddress net.IP, opts map[string]string) (*net.IPNet, map[string]string, error) {
+func (a *Allocator) RequestAddress(poolID string, prefAddress net.IP, opts map[string]string) (*net.IPNet, []string, []string, map[string]string, error) {
```

Comment:
@mavenugo: I thought through this while making the change and even I would prefer a different approach. But wanted to know how it is preferred. Would clubbing the DNS specific fields into a struct help?
If you have any other suggestions, please let me know.